src/herder/HerderImpl.cpp

```diff
@@ -399,16 +440,12 @@ HerderImpl::checkCloseTime(SCPEnvelope const& envelope, bool enforceRecent)
     // statement using consensus data:
     // update lastCloseIndex/lastCloseTime to be the highest possible but still
     // be less than envLedgerIndex
-    auto adjustLastCloseTime = [&](HerderSCPDriver::ConsensusData const* cd) {
-        if (cd && envLedgerIndex >= cd->mConsensusIndex &&
-            cd->mConsensusIndex > lastCloseIndex)
-        {
-            lastCloseIndex = static_cast<uint32>(cd->mConsensusIndex);
-            lastCloseTime = cd->mConsensusValue.closeTime;
-        }
-    };
-    adjustLastCloseTime(mHerderSCPDriver.trackingSCP());
-    adjustLastCloseTime(mHerderSCPDriver.lastTrackingSCP());
+    auto trackingIndex = trackingConsensusLedgerIndex();
```

Comment:
this is not equivalent: old code did not assume that it was tracking, this version would crash

Reply:
same thing here: I do not think this is true in the current state of the proposal as per my comment above.